File: go/easy/Two Sum.go
package easy

func twoSum(nums []int, target int) []int {
	hashNums := make(map[int]int)

	for i, e := range nums {
		s := target - e
		if index, found := hashNums[s]; found {
			return []int{i, index}
		}
		hashNums[e] = i
	}
	return []int{}
}

File: python/easy/Two Sum.py
from collections import defaultdict
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hash_nums = {v: i for i, v in enumerate(nums)}

        for i, v in enumerate(nums):
            s = target - v
            if s in hash_nums and hash_nums[s] != i:
                return [i, hash_nums[s]]
